### ISSUE TITLE
[Bug]: Redirect are not handled, so they throw errors

### ISSUE THREAD

**Issue by jakst** [solidjs/solid-start#1342]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Any `throw redirect('/login')` in a server function will cause a server error. I have also tried with `return redirect('/login')`, but that just doesn't do anything at all. I can reproduce this in a freshly created solid-start project.

![start-screenshot (1)](https://github.com/solidjs/solid-start/assets/3687893/f1986706-d2b0-453f-974a-62f0ac6a791d)

Looking at the server logs, it seems to be the actual redirect response object that is thrown as an error.

```
Error: Unknown error
    at castError (./node_modules/.pnpm/solid-js@1.8.15/node_modules/solid-js/dist/server.js:9:10)
    at ./node_modules/.pnpm/solid-js@1.8.15/node_modules/solid-js/dist/server.js:560:32 {
  [cause]: Response {
    [Symbol(realm)]: { settingsObject: {} },
    [Symbol(state)]: {
      aborted: false,
      rangeRequested: false,
      timingAllowPassed: false,
      requestIncludesCredentials: false,
      type: 'default',
      status: 302,
      timingInfo: null,
      cacheState: '',
      statusText: '',
      headersList: [HeadersList],
      urlList: []
    },
    [Symbol(headers)]: HeadersList {
      cookies: null,
      [Symbol(headers map)]: [Map],
      [Symbol(headers map sorted)]: [Array]
    }
  }
}
```

### Expected behavior 🤔

The user should be redirected to a new page.

### Steps to reproduce 🕹

Steps:

1. git clone https://github.com/jakst/solid-redirect-crash-repro
2. cd solid-redirect-crash-repro
3. pnpm install
4. pnpm dev
5. Visit [localhost:3000](localhost:3000)
6. Notice the error

**Comment by jakst** [comment]
The issue seems to have been introduced in `@solidjs/router@0.12.1`

**Comment by ryansolid** [comment]
Fixed in 0.12.4 of the router
